core/src/main/java/com/amplifyframework/auth/options/AuthSignUpOptions.java

```diff
@@ -18,60 +18,55 @@
 import androidx.annotation.NonNull;
 import androidx.core.util.ObjectsCompat;
 
+import com.amplifyframework.auth.AuthUserAttribute;
 import com.amplifyframework.util.Immutable;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Objects;
 
-public final class AuthSignUpOptions {
-    private final Map<String, String> userAttributes;
-    private final Map<String, String> validationData;
+public class AuthSignUpOptions {
+    private final List<AuthUserAttribute> userAttributes;
 
     /**
      * Advanced options for signing in.
      * @param userAttributes Additional user attributes which should be associated with this user on registration
-     * @param validationData A map of custom key/values to be sent as part of the sign up process
      */
-    private AuthSignUpOptions(
-            Map<String, String> userAttributes,
-            Map<String, String> validationData
+    protected AuthSignUpOptions(
+            List<AuthUserAttribute> userAttributes
     ) {
         this.userAttributes = userAttributes;
-        this.validationData = validationData;
     }
 
     /**
      * Get additional user attributes which should be associated with this user on registration.
      * @return additional user attributes which should be associated with this user on registration
      */
     @NonNull
-    public Map<String, String> getUserAttributes() {
+    public List<AuthUserAttribute> getUserAttributes() {
         return userAttributes;
     }
 
-    /**
-     * Get a map of custom key/values to be sent as part of the sign up process.
-     * @return a map of custom key/values to be sent as part of the sign up process
-     */
-    @NonNull
-    public Map<String, String> getValidationData() {
-        return validationData;
-    }
-
     @NonNull
-    public static Builder builder() {
-        return new Builder();
+    public static Builder<?> builder() {
```

Comment:
`Builder<? extends AuthSignUpOptions>` would be an easy improvement, if you can't achieve like a `Builder<T>`, or `B` (is a` Builder<SubClassType>`, etc.)

Sync up with @raphkim he has known these struggles in his predictions builders as well

Reply:
ahaha i'm the master of making unnecessarily complicated generic type builders now

Reply:
Ok yeah Builder<T> didn't work but that sounds like it would work and doubly enforce what already is enforced from the abstract Builder class definition.

Reply:
Hmm, so yeah this doesn't work because when you do ? extends AuthSignUpOptions.Builder it throws a raw types error because the type of T is itself a generic. I don't think it's a problem not having this since, as the online resource I got this from pointed out, the type of T is enforced in the builder class definition.